draft-ietf-httpbis-client-hints.md

```diff
@@ -184,6 +215,22 @@ Implementers ought to consider both user and server controlled mechanisms and po
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
 
+## Deployment and Security Risks
+Adding new request headers forces us to think about two aspects:
+
+  - Will user-agent emitted values impact current servers which could be using these headers for a different purpose?
+  - Can malicious content impact servers on the user's network by crafting such request headers with malicious header values?
+
+The former concern is a deployment risk, slightly mitigated by the opt-in mechanisms of Client Hints.
+
+The latter is a security concern. CORS {{FETCH}} would have helped to eleviate the latter concern, but for performance purposes, we want to make sure Client Hints request headers are considered CORS-safe.
+
+
+In order to address both of these concerns, specifications and features relying on Client Hints MUST use the "Sec-" prefix for request header names. User agents reserve this namespace for headers that can only be emitted by the browser and thus can guarantee and enforce the type and format of the data communicated in the header value fields. That restriction also means that servers are not likely to be using such prefixed-headers for other purposes.
+
+Features relying on Client Hints are also encouraged to add "CH-" to the header name, in order to make it easier to distinguish Client Hint request headers from others. Servers and intermediaries handling those headers should take special care when storing them, as they are likely to contain entropy that can contribute to user fingerprinting.
```

Comment:
@sleevi - can you take a look to see that this language reflects what you had in mind when we discussed this at the IETF105 HTTPWG meeting?

Reply:
Deployment of new request headers requires several considerations:

- Potential conflicts due to existing use of field name
- Properties and validation of communicated data in field value

Specifications and features relying on Client Hints MUST use the "Sec-" prefix for request header names. User agents reserve this namespace that can only be emitted by the browser and thus can guarantee and enforce the type and format of the data communicated in the header value fields. Further, specifications and features relying on Client Hints are encouraged to include "CH-" in the header name, in order to make it easier to distinguish Client Hint request headers from others. 

The combination of the above recommendations and requirements, plus the opt-in mechanism of Client Hints, helps minimize deployment risks. Further, these requirements enable server and intermediaries handling requests with Client Hints to identify such request headers and take precautions in how they are used and processed — e.g. opt out from storing them as they may expose information about the user.